rust/connlib/clients/android/src/lib.rs

```diff
@@ -1,43 +1,44 @@
-#[macro_use]
-extern crate log;
-extern crate android_logger;
-extern crate jni;
-use self::jni::JNIEnv;
-use android_logger::Config;
+// The "system" ABI is only needed for Java FFI on Win32, not Android:
+// https://github.com/jni-rs/jni-rs/pull/22
+// However, this consideration has made it idiomatic for Java FFI in the Rust
+// ecosystem, so it's used here for consistency.
+
 use firezone_client_connlib::{
     Callbacks, Error, ErrorType, ResourceList, Session, TunnelAddresses,
 };
-use jni::objects::{JClass, JObject, JString, JValue};
-use log::LevelFilter;
+use jni::{
+    objects::{JClass, JObject, JString, JValue},
+    JNIEnv,
+};
 
 /// This should be called once after the library is loaded by the system.
 #[allow(non_snake_case)]
 #[no_mangle]
 pub extern "system" fn Java_dev_firezone_connlib_Logger_init(_: JNIEnv, _: JClass) {
-    #[cfg(debug_assertions)]
-    let level = LevelFilter::Trace;
-    #[cfg(not(debug_assertions))]
-    let level = LevelFilter::Warn;
-
     android_logger::init_once(
-        Config::default()
-            // Allow all log levels
-            .with_max_level(level)
+        android_logger::Config::default()
+            .with_max_level(if cfg!(debug_assertions) {
+                log::LevelFilter::Trace
+            } else {
+                log::LevelFilter::Warn
+            })
             .with_tag("connlib"),
     )
 }
 
-pub enum CallbackHandler {}
+#[derive(Clone)]
+pub struct CallbackHandler;
+
 impl Callbacks for CallbackHandler {
-    fn on_update_resources(_resource_list: ResourceList) {
+    fn on_update_resources(&self, _resource_list: ResourceList) {
         todo!()
     }
 
-    fn on_set_tunnel_adresses(_tunnel_addresses: TunnelAddresses) {
+    fn on_set_tunnel_adresses(&self, _tunnel_addresses: TunnelAddresses) {
```

Comment:
We [may have renamed](https://github.com/firezone/product/blob/main/architecture/madr.md#2023_06_19-grooming) this callback to `onConnect` in the last grooming. Will defer to you though -- I don't feel strongly Re: naming conventions

Reply:
Absolutely; I was trying to avoid adding something else onto an already large-ish PR, but that's a small enough addition that it'd be worth me adding now just to avoid additional churn for the clients.

Reply:
Ah gotcha!